lib/astarte/client/appengine/devices.ex

```diff
@@ -73,7 +73,7 @@ defmodule Astarte.Client.AppEngine.Devices do
     request_path = "devices/#{device_id}/interfaces/#{interface}#{path}"
 
     with {:ok, %Tesla.Env{} = result} <- Tesla.delete(tesla_client, request_path) do
-      if result.status == 200 do
```

Comment:
That's actually unrelated with adding tests - it should go at least in a separate commit, probably in a separate MR